### ISSUE TITLE
CFLint not working 

### ISSUE THREAD

**Issue by ToombaAlexander** [SublimeLinter/SublimeLinter#1491]
Hi,

I've packages:

* CFML: 0.27.4
* Sublime​Linter: 4.8.0
* Sublime​Linter-contrib-CFLint: 3.0.0

And CFLint-1.4.0-all.jar ("jar_file": "CFLint-1.4.0-all.jar") in my settings of SublimeLinter

**But it doesn't seems to work. If i do a wrong syntax error or whatever i don't see any messages in sublime.**

When i open the console in sublime and open a CFC i get the following message, and its not working (See below)
Also if i open the Sublimelinter output (you can find this leftbottom of sublime) it only says: No lint results.

> SublimeLinter: sublime_linter.py:250: Delay buffer 69 for 0.0s
> SublimeLinter: sublime_linter.py:250: Delay buffer 76 for 0.0s
> SublimeLinter: linter.py:887: Checking lint mode load_save vs lint reason on_load
> SublimeLinter: #1 linter.py:907: 'cflint' is linting 'test.cfc'
> 
> SublimeLinter: #1 linter.py:1191: The `@` symbol in cmd has been deprecated. Use $file, $temp_file or $file_on_disk instead.
> SublimeLinter: #1 linter.py:1263: Running ...
> 
>   C:\websites\test  (working dir)
>   > "C:\Program Files (x86)\Common Files\Oracle\Java\javapath\java.exe" -file test.cfc -q -text
> 
> SublimeLinter: sublime_linter.py:444: Linting buffer 76 took 0.64s

On save:

> SublimeLinter: linter.py:887: Checking lint mode load_save vs lint reason on_save
> SublimeLinter: #2 linter.py:907: 'cflint' is linting 'test.cfc'
> SublimeLinter: #2 linter.py:1191: The `@` symbol in cmd has been deprecated. Use $file, $temp_file or $file_on_disk instead.
> SublimeLinter: #2 linter.py:1263: Running ...
> 
>   C:\websites\test  (working dir)
>   > "C:\Program Files (x86)\Common Files\Oracle\Java\javapath\java.exe" -file test.cfc -q -text
> 
> SublimeLinter: sublime_linter.py:444: Linting buffer 76 took 0.09s

**Comment by IperStan** [comment]
> 🤔 I already answered this one [SublimeLinter/SublimeLinter-flow#69 (comment)](https://github.com/SublimeLinter/SublimeLinter-flow/issues/69#issuecomment-420002470).
> 
> > And CFLint-1.4.0-all.jar ("jar_file": "CFLint-1.4.0-all.jar") in my settings of SublimeLinter
> 
> > the settings key is just `jar` (not `jar_file`). (It's also just `configfile` instead of `config_file_name`.) Probably the maintainer of the repo never changed the README. You can try a PR for the README over there.

Hi, thanks for the explicit note about the outdated README.
Now, though, I have another problem with showing the results of linting process.

My setup is:
SublimeText: 3.3143
CFML: 0.27.4
Sublime​Linter: 4.8.0
Sublime​Linter-contrib-CFLint: 3.0.0

It seems that after generating the report file of linting results, SublimeLinter just "forgets" to parse it.

This is the output in SublimeText console with "debug": true in the SublimeLinter config:

SublimeLinter: SublimeLinter.sublime-package:887: Checking lint mode background vs lint reason on_save
SublimeLinter: #18 SublimeLinter.sublime-package:907: 'cflint' is linting 'testLint.cfc'
SublimeLinter: #18 SublimeLinter.sublime-package:1191: The `@` symbol in cmd has been deprecated. Use $file, $temp_file or $file_on_disk instead.
SublimeLinter: #18 SublimeLinter.sublime-package:1263: Running ...

  C:\Projects\testLinting  (working dir)
  > "C:\Program Files (x86)\Common Files\Oracle\Java\javapath\java.exe" -jar C:/Projects/CFLint-1.4.0-all.jar -file C:\Projects\testLintig\testLint.cfc -q -text

SublimeLinter: SublimeLinter.sublime-package:444: Linting buffer 39 took 3.27s

Is there any missing piece in config file to explain linter that the file "cflint-result.txt" has to be parsed and therefore show the errors inside the SublimeText?

**Comment by kaste** [comment]
You should not redirect output to a file so it goes to stdout. Maybe you define output in your config file?

**Comment by IperStan** [comment]
> Maybe you define output in your config file?

I haven't created any config file so far due to this first launch problem. Hence, I'm only running the basic setup with default rules of CFLinter.

>You should not redirect output to a file so it goes to stdout. 

I considered it as a potential issue but since in the source file of the SublimeLinter-contrib-CFLint [linter.py](https://github.com/ckaznocha/SublimeLinter-contrib-CFLint/blob/master/linter.py#L19):
I didn't find out a solution to exclude "-text" argument and override it with "-stdout".

Could you kindly explain what in my case shall I do to redirect output to stdout?

**Comment by kaste** [comment]
Hm, it's getting funky. I *think* you should add stdout and not replace `-text`. So, it would be (in the SublimeLinter settings)

```
	"cflint": {
		"args": "-stdout"
	}
```

**Comment by IperStan** [comment]
> Hm, it's getting funky. I _think_ you should add stdout and not replace `-text`. So, it would be (in the SublimeLinter settings)
> 
> ```
> 	"cflint": {
> 		"args": "-stdout"
> 	}
> ```

In this way linter runs with the following command:

SublimeLinter: SublimeLinter.sublime-package:887: Checking lint mode background vs lint reason on_save
SublimeLinter: #35 SublimeLinter.sublime-package:907: 'cflint' is linting 'testLint.cfc'
SublimeLinter: #35 SublimeLinter.sublime-package:1191: The `@` symbol in cmd has been deprecated. Use $file, $temp_file or $file_on_disk instead.
SublimeLinter: #35 SublimeLinter.sublime-package:1263: Running ...

  C:\Projects\testLinting  (working dir)
  > "C:\Program Files (x86)\Common Files\Oracle\Java\javapath\java.exe" -stdout -jar C:/Projects/CFLint-1.4.0-all.jar -file C:\Projects\testLinting\testLint.cfc -q -text

SublimeLinter: SublimeLinter.sublime-package:444: Linting buffer 452 took **0.06s**

Notice that this time it took literally nothing **(0.06s)** to lint since it didn't create any file but put it straight in stdout as expected, BUT, the linter haven't shown any higlights or icons in SublimeText.

It seems SublimeLinter still doens't "catch" the output.

**Comment by kaste** [comment]
Could you go to the command line and try the commands SublimeLinter executes here? SublimeLinter should output stdout to the Sublime console in debug mode so it seems there isn't any output.

**Comment by IperStan** [comment]
> Hm, maybe your default config file doesn't include any rules? You could create a config file and point to it via -configfile.

In fact I don't have any config file at all because in docs of CFLint they point out there is [a global config](https://github.com/cflint/CFLint#global-configuration)
and it looks that everything works fine with CFLinter, I have tried to execute within command line and it generates plenty of "good" output. Once I tried to redirect this to the human readable UI in sublimeText it just doesn't show anything.

To sum up, I just have this folder structure:
C:\Projects\\:
.\CFLint-1.4.0-all.jar
--------------\testLinting\
---------------------------.\testLint.cfc (with bunch of errors for test purpose)

**Comment by IperStan** [comment]
**TEMPORARY SOLUTION FOUND!**
All right I found out a workaround based on the info you provided me.

Since the problem of putting "-stdout" in the right place wasn't resolvable with "args" property inside SublimeLinter config, I just cut "jar"'s property content and put it inside "args" and THEN put "-stdout" argument.

So with this config SublimeLinter finally shows errors in SublimeText:

SublimeLinter.sublime-settings content:
`{
  "linters": {
    "cflint": {
      "disable": false,
      "args": "-jar C:/Progetti/CFLint-1.4.0-all.jar -stdout"
    }
  }
}`


I don't see any solution to remove "-text" since it's useless and might confuse many others.
Any ideas?

Thanks anyway for helping!

**Comment by kaste** [comment]
`-text` is mandatory otherwise it outputs html to stdout.

I can confirm that this plugin is broken. I patched locally and found that

```
	return ['java', '-jar', jar, '${args}', '-stdin', '${file}', '-q', '-text', '-stdout']
```

works very well. Note that I also use `-stdin` instead of `-file` and by that enable background linting.

**Comment by kaste** [comment]
I made a PR! https://github.com/ckaznocha/SublimeLinter-contrib-CFLint/pull/27

**Comment by kaste** [comment]
The PR has been merged, so I'm closing. @IperStan 	I think you need to revert the config hack as soon as you get the new version.
